pkg/httpproxy/sign.go

```diff
@@ -115,8 +116,13 @@ func (a awsv4) getServiceAndRegion(host string) (string, string) {
 		}
 	}
 
-	// if no region is found, global endpoint is assumed. In this case us-east-1 should be used for signing:
+	// if no region is found, global endpoint is assumed.
 	// https://docs.aws.amazon.com/general/latest/gr/sigv4_elements.html
+	if strings.Contains(host, "us-gov") {
+		// For us-gov, the default region is us-gov-west-1
+		return service, defaultUSGovAWSRegion
+	}
+	// In this case us-east-1 should be used for signing
```

Comment:
nit: ik I wrote it originally but could remove mention of us-gov-west-1 and us-east-1 in comments since if the values change they will be outdated, plus the docs are linked for reference which should be sufficient.

Reply:
Done